sof_wrapper/api/views.py

```diff
@@ -82,3 +78,81 @@ def auditlog_addevent():
     log_level_method = getattr(logging, level.lower())
     log_level_method(body)
     return jsonify(message='ok')
+
+
+@base_blueprint.route('/save_data', methods=('POST',))
+def save_data():
+    """Write JSON to disk
+
+    API for client applications to write JSON directly to disk.
+    Intended use: front-end can capture data at any point and write
+    to disk for testing & verification.
+
+    Returns a json friendly message, i.e. {"message": "ok"} or details on error
+    ---
+    operationId: save_data
+    tags:
+      - debugging
+    produces:
+      - application/json
+    parameters:
+      - in: body
+        name: body
+        schema:
+          id: save_data
+          required:
+            - context
+            - data
+          properties:
+            filename:
+              type: string
+              description: string to save file as.  UUID used by default.
+                does not allow overwrites.
+            context:
+              type: string
+              description: context such as "CQL MME calc"
+            data:
+              type: json
+              description: json serialized data to capture
+    responses:
+      200:
+        description: successful operation
+        schema:
+          id: response_ok
+          required:
+            - message
+          properties:
+            message:
+              type: string
+              description: Result, typically "ok"
+
+    """
+    body = request.get_json()
```

Comment:
I'd prefer this only enabled on dev environments. We should really work on building validation into CI if we don't want to be manually testing all the time.
```suggestion
    if app.config["ENV"] != "development:
        return jsonify(message="Disabled on non-dev deploys"), 401
    body = request.get_json()
```